src/lock_file/mod.rs

```diff
@@ -0,0 +1,521 @@
+mod python;
+mod python_name_mapping;
+
+use crate::{progress, Project};
+use futures::TryStreamExt;
+use futures::{stream, StreamExt};
+use indicatif::ProgressBar;
+use itertools::Itertools;
+use miette::{Context, IntoDiagnostic};
+use rattler_conda_types::{
+    Channel, GenericVirtualPackage, MatchSpec, NamelessMatchSpec, PackageName, Platform,
+    RepoDataRecord, Version,
+};
+use rattler_lock::{
+    builder::{
+        CondaLockedDependencyBuilder, LockFileBuilder, LockedPackagesBuilder,
+        PipLockedDependencyBuilder,
+    },
+    CondaLock, LockedDependency, PackageHashes,
+};
+use rattler_repodata_gateway::sparse::SparseRepoData;
+use rattler_solve::{resolvo, SolverImpl};
+use rip::Wheel;
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    str::FromStr,
+    sync::Arc,
+    time::Duration,
+};
+
+/// Loads the lockfile for the specified project or returns a dummy one if none could be found.
+pub async fn load_lock_file(project: &Project) -> miette::Result<CondaLock> {
+    let lock_file_path = project.lock_file_path();
+    tokio::task::spawn_blocking(move || {
+        if lock_file_path.is_file() {
+            CondaLock::from_path(&lock_file_path).into_diagnostic()
+        } else {
+            LockFileBuilder::default().build().into_diagnostic()
+        }
+    })
+    .await
+    .unwrap_or_else(|e| Err(e).into_diagnostic())
+}
+
+/// Returns true if the locked packages match the dependencies in the project.
+pub fn lock_file_up_to_date(project: &Project, lock_file: &CondaLock) -> miette::Result<bool> {
+    let platforms = project.platforms();
+
+    // TODO: Add support for python dependencies
+    if !project.python_dependencies().is_empty() {
```

Comment:
We should also check here if there are pip managed dependencies in the lockfile to regenerate for removal of any dependencies. 

As going from x to 0 python dependencies doesn't trigger the update.

Reply:
Lets do in another PR when we integrate the lock-file better

Reply:
Done! https://github.com/prefix-dev/pixi/issues/467